repo/blob/s3/s3_storage.go

```diff
@@ -312,6 +312,42 @@ func (s *s3Storage) FlushCaches(ctx context.Context) error {
 	return nil
 }
 
+func isAccessDenied(err error) bool {
+	var e minio.ErrorResponse
+	return errors.As(err, &e) && e.Code == "AccessDenied"
+}
+
+func (s *s3Storage) bucketExists(ctx context.Context) (bool, error) {
+	ok, err := s.cli.BucketExists(ctx, s.Options.BucketName)
+	if err == nil {
+		return ok, nil
+	}
+
```

Comment:
Check here whether or not the bucket exists, that is:
- when `err == nil && ok`, then `return nil`
- when `err == nil && !ok`, then `return errors.Errorf("bucket %q does not exist", bucketName)`.

Essentially, move here the check that was performed in the caller.

Reply:
Done, thanks